esrally/utils/net.py

```diff
@@ -70,10 +70,22 @@ def finish(self):
         self.p.finish()
 
 
+def _fake_import_boto3():
+    # This function only exists to be mocked in tests to raise an ImportError, in
+    # order to simulate the absence of boto3
```

Comment:
This is quite ugly, but the alternative is to use `del sys.modules["boto3"]`, which is an ugly and global hack, and likely to break things. While we use this hack succesfully in urllib3 to test support for Python versions without ssl, this:

 * has caused [huge headaches in a project that adds async support to urllib3](https://github.com/python-trio/hip/pull/42#issuecomment-386304273) and
 * would prevent running tests in parallel.

So I would suggest avoiding touching `sys.modules`.